### PR TITLE
Populate application list from server

### PR DESCRIPTION
**Pull request by varkor**
Also add a few options to filter the data.

### CODE REVIEW

File: src/application-listing/ApplicationListing.js [hunk 1]
```diff
@@ -1,43 +1,91 @@
...
+  constructor (props) {
+    super(props);
+    this.state = {
+      applications: [],
+      ratingCutoff: 0,
+      ratingComparison: 1,
+      gender: null,
+      inTeam: null,
+    };
+    api.getApplications().then(applications => {
```
Comment (Pinpickle) [inline src/application-listing/ApplicationListing.js]: How is this working without an access token?

Comment (varkor) [reply src/application-listing/ApplicationListing.js]: Oops, because I had an extension that was automatically populating the Authorization header.

**Review by jaredkhan — COMMENTED** [review]
I'm pretty much happy with this it just needs links to the applications / application ID listing to make it useful!

**Review by jaredkhan — APPROVED** [review]
LGTM

File: src/application-listing/ApplicationList.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { Form } from 'reactstrap';
+import Table from 'rc-table';
+import './application-listing.css';
+
+class ApplicationList extends React.Component {
+  constructor (props) {
+    super(props);
+    const { applications } = props;
```
Comment (Pinpickle) [inline src/application-listing/ApplicationList.js]: If this component ever receives new applications as props for some reason, the component won't update as we're duplicating applications in state. Is there any reason why we can't use applications in props the whole time?

Comment (varkor) [reply src/application-listing/ApplicationList.js]: Nope, this is a good point. Fixed.

**Review by Pinpickle — APPROVED** [review]
Happy for this to ship once the required API endpoints are available!

File: src/application-listing/ApplicationList.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          <label>
+            In a team:
+            <select onChange={ event => this.filter({ inTeam: event.target.value !== "null" ? event.target.value === "Yes" : null }) }>
+              <option value="null">Either</option>
+              <option>Yes</option>
+              <option>No</option>
+            </select>
+          </label>
+        </Form>
+        <Table className="table" columns={columns} data={displayApplications} emptyText={() => <span className="empty-table">No applications match the current filter</span>} onRowClick={(record, index) => window.location = `applications/${record.id}`} />
```
Comment (Pinpickle) [inline src/application-listing/ApplicationList.js:75]: We can avoid a reload of the page by dispatching a push action to the store ([example](https://github.com/hackcambridge/hc-applications/blob/master/src/application/reducer.js#L45)), or if possible, using `Link` elements from react-router ([example](https://github.com/hackcambridge/hc-applications/blob/master/src/header/Header.js#L8)). The latter is preferred as cmd+click/"open link in new tab" will still work to open in new tab

**Pull request merged**